source/computer/commands.rst

```diff
@@ -66,51 +66,688 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
+
+如果不确定该命令执行后会产生什么结果，可以先用 ``-n`` 选项模拟执行的结果，并不真的执行命令::
+
+    $ rsync -anv ~/Downloads/source ~/workspace/destination
+
+如果只想同步源目录 :file:`~/Downloads/source` 里面的内容到目标目录 :file:`~/workspace/destination` 中，
+则需要在源目录后面加上斜杠。此时，目标目录中不会产生 :file:`~/workspace/source` 目录::
+
+    $ rsync -av ~/Downloads/source/ ~/workspace/destination
+
+默认情况下，该命令只确保源目录 :file:`~/Downloads/source/` 里的所有内容都复制到目标目录中。
+它不会使两个目录保持相同，并且不会删除目标目录中的文件和目录。如果要使源目录和目标完全同步，
+则可以使用 ``--delete`` 选项删除目标目录中不存在于源目录的文件和目录。
+此时，目标目录 :file:`~/workspace/destination` 成为源目录 :file:`~/Downloads/source/`
+的一个镜像::
+
+    $ rsync -av --delete ~/Downloads/source/ ~/workspace/destination
+
+.. note::
+
+   如果源目录后面没有加斜杠，即::
+
+       $ rsync -av --delete ~/Downloads/source ~/workspace/destination
+    
+   则 ``--delete`` 选项只会保证目标目录中的 :file:`~/workspace/destination/source` 目录
+   与源目录 :file:`~/Downloads/source` 同步。目标目录中的其他目录和文件
+   （如 :file:`~/workspace/test`\ 、\ :file:`~/workspace/README.md`\ ）并不会被删除。
+   此时，源目录 :file:`~/Downloads/source` 的镜像是 :file:`~/workspace/destination/source`\ 。
+
+使用该命令在本地计算机与远程计算机之间同步文件时，用法相同，只需在源目录或者目标目录前加上
+``username@remote_host:`` 表示远程计算机即可。其中 ``username`` 表示用户帐号，``remote_host``
+表示远程计算 IP 地址。``rsync`` 默认使用 `ssh`_ 进行远程登录和数据传输。
+以下命令假定远程电脑的 IP 地址是 192.168.1.100，用户名是 seismo-learn。
+
+将本地源目录 :file:`~/Downloads/source` 同步到远程计算机的目标目录 :file:`~/destination` 中::
+
+    $ rsync -av --delete ~/Downloads/source seismo-learn@192.168.1.100:~/destination
+
+将远程计算机的源目录 :file:`~/source` 同步到本地目标目录 :file:`~/workspace/destination` 中::
+
+    $ rsync -av --delete seismo-learn@192.168.1.100:~/source ~/workspace/destination
+
 scp
 ---
 
+``scp`` 命令的命名来源于 **s**\ ecure **c**\ o\ **p**\ ，可用于本地和远程电脑之间传输文件。
+该命令基于 `ssh`_ 进行安全的远程文件传输，因此传输是加密的。虽然 ``scp`` 传输速度不如 `rsync`_
+命令，但是它不占资源，不会提高多少系统负荷。当有许多小文件需要传输时，使用 `rsync`_ 命名会导致
+硬盘 I/O（输入/输出）非常高，而 ``scp`` 基本不影响系统正常使用。
+
+以下命令假定远程电脑的 IP 地址是 192.168.1.100，用户名是 seismo-learn。
+
+复制远程文件 :file:`/home/seismo-learn/fk3.3.tar.gz` 到本地目录 :file:`~/Downloads` 下::
+
+    $ scp seismo-learn@192.168.1.100:/home/seismo-learn/fk3.3.tar.gz ~/Downloads/
+
+复制远程目录 :file:`/home/seismo-learn/folder1` 到本地目录 :file:`~/Downloads` 下::
+
+    $ scp -r seismo-learn@192.168.1.100:/home/seismo-learn/folder1 ~/Downloads/
+
+上传本地文件 :file:`~/Downloads/fk3.3.tar.gz` 到远程目录 :file:`home/seismo-learn/folder2` 下::
+
+    $ scp ~/Downloads/fk3.3.tar.gz seismo-learn@192.168.1.100:/home/seismo-learn/folder2/
+
+上传本地目录 :file:`~/Downloads/folder1` 到远程目录 :file:`home/seismo-learn/folder2` 下::
+
+    $ scp ~/Downloads/folder1 seismo-learn@192.168.1.100:/home/seismo-learn/folder2/
+
+sed
+---
+
+``sed`` 命令的名字来源与 **s**\ tream **ed**\ itor（流编辑器）。该命令可以用于对输入流
+（文件或管道）执行基本的文本转换。它会把当前处理的行存储在临时缓冲区中再进行处理，处理完成后
+再把缓冲区的内容送往屏幕。接着处理下一行，直到文件末尾。因此默认情况下，文件内容并没有改变。
+
+将 :file:`file` 中每一行的第一个 book 替换成 books::
+
+    $ sed 's/book/books/' file
+
+将 :file:`file` 中每一行的所有的 book 都替换成 books::
+
+    $ sed 's/book/books/g' file
+
+以上命令只是将转换后的文本内容打印出来，并为改变文件本身。可以使用 ``-i`` 选项直接改变文件::
+
+    $ sed -i 's/book/books/g' file
```

Comment:
This doesn't works with BSD sed.

Reply:
As we discussed in #91, we may use GNU.

Reply:
I believe most people just run `sed -i` and are confused that it reports a weird error on macOS.

Reply:
I think most Chinese seismologists use Linux.

Reply:
We recommend macOS users try GNU, right?

Reply:
Most users won't remember to add the `g` prefix.

Reply:
I think this section focus on Linux commands, while this issue is related to BSD and GNU commands, right?